File: src/main/kotlin/br/com/fiap/techfood/application/ports/outbound/OrderOutboundPort.java
package br.com.fiap.techfood.application.ports.outbound;

import br.com.fiap.techfood.application.core.domains.OrderDomain;
import br.com.fiap.techfood.application.core.domains.enums.OrderStatusEnum;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface OrderOutboundPort {

    OrderDomain save(OrderDomain orderDomain);

    Optional<OrderDomain> findById(UUID id);

    List<OrderDomain> findAllByOrderStatus(OrderStatusEnum status);

    void delete(UUID id);

    //CHECAR QUAIS PARAMS ID?
    void approvePayment(UUID id);

}

File: src/main/kotlin/br/com/fiap/techfood/application/ports/outbound/OrderOutboundPort.kt
package br.com.fiap.techfood.application.ports.outbound

import br.com.fiap.techfood.application.core.domains.OrderDomain
import br.com.fiap.techfood.application.core.domains.enums.OrderStatusEnum
import java.util.*

interface OrderOutboundPort {

    fun save(orderDomain: OrderDomain): OrderDomain

    fun findById(id: UUID): Optional<OrderDomain>

    fun findAllByOrderStatus(status: OrderStatusEnum): List<OrderDomain>

    fun delete(id: UUID)

    //CHECAR QUAIS PARAMS ID?
    fun updateStatus(id: UUID, status: OrderStatusEnum)
}
